pubtools/iib/iib_ops.py

```diff
@@ -237,6 +244,61 @@ def _iib_op_main(
     return build_details
 
 
+def run_iib_build_with_retries(
+    iiblib_method: Callable[[], IIBBuildDetailsModel],
+    pc: pushcollector.Collector,
+    iib_client: IIBClient,
+    iib_server: str,
+    retries: int = 1,
+) -> IIBBuildDetailsModel:
+    """
+    Run the IIB build and retry if a specific error types occur.
+
+    Args:
+        iiblib_method (Callable):
+            iiblib method to be called. The method should be callable with no parameters.
+        pc: (Collector):
+            PushCollector instance.
+        iib_client (IIBClient):
+            iiblib client instance.
+        iib_server (str):
+            IIB server.
+        retries (int, optional):
+            Maximum number of times to retry. Defaults to 1.
+    Returns (IIBBuildDetailsModel):
+        Build details as returned by iiblib.
+    """
+    build_details = iiblib_method()
+
+    push_items = push_items_from_build(build_details, "PENDING")
+    LOG.debug("Updating push items")
+    pc.update_push_items(push_items)
+
+    build_details_url = _make_iib_build_details_url(iib_server, build_details.id)
+    LOG.info("IIB details: %s", build_details_url)
+
+    build_details = iib_client.wait_for_build(build_details)
+    # If state_reason doesn't match any retirable reason
+    if not any([build_details.state_reason.startswith(r) for r in RETRY_REASONS]):
+        return build_details
+
+    for i in range(retries):
+        LOG.info(
+            f"IIB build failed for retirable reason: {build_details.state_reason}. "
+            f"Attempting retry {i + 1}/{retries}"
+        )
+
+        build_details = iiblib_method()
```

Comment:
In my test, the retried iib request was failed with 401. Seems the negotiation token cannot be reused in later requests. I'm wondering if it's worth to fix the retry as it never worked.

Reply:
I agree, I don't think implementing the retries is worth it at this point. I'm closing this PR. The iiblib PR should fix the issue observed in the Jira.